app/Http/Controllers/RegistrationController.php

```diff
@@ -77,13 +94,50 @@ public function store(Request $request)
       return redirect()->route('profile.create')->with('flash_message', ['text' => 'Thanks for creating your account.', 'class' => '-success']);
   }
 
-    public function sendRegistrationEmail($user)
-    {
-        $email = new Email();
-        $data = [
-      'first_name' => $user->first_name,
-      'last_name'  => $user->last_name,
-    ];
-        $email->sendEmail('welcome', 'applicant', $user->email, $data);
-    }
+  public function sendRegistrationEmail($user)
+  {
+      $email = new Email();
+      $data = [
+        'first_name' => $user->first_name,
+        'last_name'  => $user->last_name,
+      ];
+      $email->sendEmail('welcome', 'applicant', $user->email, $data);
+  }
+
+  /**
+   * Show the form for editing the specified resource.
+   *
+   * @param  int  $id
+   *
+   * @return Response
+   */
+  public function edit($id)
+  {
+      $user = User::whereId($id)->firstOrFail();
+
+      return view('registration.edit')->withUser($user);
+  }
+
+  /**
+   * Update the specified resource in storage.
+   * PUT /registration/{id}.
+   *
+   * @param Request $request
+   * @param Page $page
+   * @return \Illuminate\Http\RedirectResponse
+   */
+  public function update(Request $request, User $user)
+  {
+      $this->validate($request, $this->rules, $this->messages);
+
+      $user =  User::whereId(Auth::user()->id)->firstOrFail();
```

Comment:
oh my goodness. this is the kind of stuff that happens when I've been looking at this all day 😬